### ISSUE TITLE
[BUG]: particle_cspt.py fails if pyp.scratch points to an NFS mountpoint

### ISSUE THREAD

**Issue by jpellman** [nextpyp/pyp#1]
Hi,

Just FYI, `particle_cspt.py` will throw the following error if `pyp.scratch` is an NFS mount:

```
pyp/refine/csp/particle_cspt.py:57 | Total time elapsed (csp_local_merge): 00h 00m 00s
57 Traceback (most recent call last):
58 File "/opt/pyp/bin/run/pyp", line 3965, in <module>
59 particle_cspt.merge_movie_files_in_job_arr(
60 File "/opt/pyp/src/pyp/utils/timer.py", line 78, in wrapper_timer
61 return func(*args, **kwargs)
62 File "/opt/pyp/src/pyp/refine/csp/particle_cspt.py", line 538, in merge_movie_files_in_job_arr
63 run_reconstruction(
64 File "/opt/pyp/src/pyp/utils/timer.py", line 78, in wrapper_timer
65 return func(*args, **kwargs)
66 File "/opt/pyp/src/pyp/refine/csp/particle_cspt.py", line 780, in run_reconstruction
67 par_obj = call_shape_phase_residuals(
68 File "/opt/pyp/src/pyp/utils/timer.py", line 78, in wrapper_timer
69 return func(*args, **kwargs)
70 File "/opt/pyp/src/pyp/analysis/scores.py", line 798, in call_shape_phase_residuals
71 par_obj = shape_phase_residuals(
72 File "/opt/pyp/src/pyp/analysis/scores.py", line 354, in shape_phase_residuals
73 results = manager.Queue()
74 File "/usr/local/envs/pyp/lib/python3.8/multiprocessing/managers.py", line 740, in temp
75 token, exp = self._create(typeid, *args, **kwds)
76 File "/usr/local/envs/pyp/lib/python3.8/multiprocessing/managers.py", line 623, in _create
77 conn = self._Client(self._address, authkey=self._authkey)
78 File "/usr/local/envs/pyp/lib/python3.8/multiprocessing/connection.py", line 502, in Client
79 c = SocketClient(address)
80 File "/usr/local/envs/pyp/lib/python3.8/multiprocessing/connection.py", line 630, in SocketClient
81 s.connect(address)
82 PermissionError: [Errno 13] Permission denied
83 2023-11-14 11:18:09 [ERROR] /opt/pyp/bin/run/pyp:3985 | PYP (csp_local_merge) failed
```

This is because `TMPDIR` is being set to a path under `pyp.scratch` [here](https://github.com/nextpyp/pyp/blob/44831379e534633d9e16b1bd63c5ba1415d691ce/src/pyp_main.py#L3298).  By default, Python's `multiprocessing` library tries to create IPC/Unix sockets under `TMPDIR`.  On an NFS volume, however, any sockets that are created will be read-only and `multiprocessing` will be unable to pass messages by writing to the socket.

My workaround was to create a symlink to `/tmp` under `pyp.scratch` (_/common/sw/nextpyp/tmp_):

```
jpellman@memc-gpu03:/common/sw/nextpyp/tmp$ sudo ln -s /tmp nextpyp_TMPDIR
jpellman@memc-gpu03:/common/sw/nextpyp/tmp$ ls -l
total 168
drwxr-xr-x  2 nextpyp nextpyp  4096 Nov 15 09:49 976
drwxr-xr-x  3 nextpyp nextpyp  4096 Nov 15 09:50 978
lrwxrwxrwx  1 root    root        4 Nov 15 13:07 nextpyp_TMPDIR -> /tmp
drwxr-xr-x 42 nextpyp nextpyp 16384 Nov 15 09:50 nextpyp_TMPDIR.bak
```

The motivation for `pyp.scratch` being set to an NFS mount was that we ran out of local temp space when we were testing out and jobs were failing.  The current documentation states that `pyp.scratch` should _ideally_ be local storage, but this socket issue means that it _necessarily_ has to be local storage (unless the code is changed so that `TMPDIR` isn't modified).

--John

**Comment by abartesaghi** [comment]
Sorry @jpellman, my github notifications were turned off and I never saw this issue until now. The next release will include a fix for this. Thanks for reporting.

**Comment by abartesaghi** [comment]
This issue was addressed in version [0.6.1](https://github.com/nextpyp/pyp/releases/tag/0.6.1).
